src/precice/config/ParticipantConfiguration.cpp

```diff
@@ -616,6 +608,23 @@ void ParticipantConfiguration:: finishParticipantConfiguration
     participant->addWatchPoint ( watchPoint );
   }
   _watchPointConfigs.clear ();
+
+
+  // create default master communication if needed
+  if(context.size > 1 && not _isParallelSolutionDefined && participant->getName() == context.name){
+    #ifdef PRECICE_NO_MPI
+        std::ostringstream error;
+        error << "When building with \"mpi=off\", you need to specify an alternative \"master\" "
+              << "communication (e.g. sockets) for every parallel participant";
+        throw std::runtime_error{error.str()};
```

Comment:
You can collapse those lines into a single throw:
```cpp
throw std::runtime_error{ "When building with \"mpi=off\", you need to specify an alternative \"master\" "
              "communication (e.g. sockets) for every parallel participant"};
```

Note that following `"c" " string " "literals"` will be joined by the compiler.

Reply:
Yes, done.